lib/Myriad/RPC/Client/Implementation/Perl.pm

```diff
@@ -26,7 +27,12 @@ method configure (%args) {
     $transport = delete $args{transport} if $args{transport};
 }
 
+method is_started() {
+    return defined $started ? $started : Myriad::Exception::InternalError->new(message => '->start was not called')->throw;
```

Comment:
I'm not a big fan of this, why we can't just await on `start`?

Reply:
the behaviour I was facing is this. 
keep in mind that we call this `start` without awaiting instead we use `retain` because if we call await, it will keep blocking as its subscribing and that subscribe future will only be done when subscription is finished.

so when we call retain without await, we end up with returning control back before `start` is finished, which will cause us to request RPC method before we actually subscribe to response channel.

Hence added this future, that will be done when we are done with the subscribing step, while subscription future will stay unresolved as long as we subscribed

Reply:
This case can be handled in `rpc_client` specifically? I slightly unable to see the benefit here